tool/site/client-react/src/components/claim/createDispenseClaim.ts

```diff
@@ -108,67 +109,70 @@ function createClaimItem(
   medicationDispenses: Array<fhir.MedicationDispense>,
   claimFormValues: ClaimFormValues
 ): fhir.ClaimItem {
-  const lineItemIds = medicationRequests.map(getMedicationRequestLineItemId)
-
-  const exemptionStatusCodeableConcept: fhir.CodeableConcept = {
-    coding: VALUE_SET_PRESCRIPTION_CHARGE_EXEMPTION.filter(coding => coding.code === claimFormValues.exemption.code)
-  }
-
   return {
     extension: [prescriptionStatusExtension],
     sequence: 1,
     productOrService: CODEABLE_CONCEPT_PRESCRIPTION,
-    programCode: [
-      exemptionStatusCodeableConcept,
-      claimFormValues.exemption.evidenceSeen
-        ? CODEABLE_CONCEPT_EXEMPTION_EVIDENCE_SEEN
-        : CODEABLE_CONCEPT_EXEMPTION_NO_EVIDENCE_SEEN
-    ],
-    detail: lineItemIds.map((lineItemId, index) => {
-      const medicationRequestForLineItem = medicationRequests.find(
-        medicationRequest => getMedicationRequestLineItemId(medicationRequest) === lineItemId
-      )
-      const medicationDispensesForLineItem = medicationDispenses.filter(
+    programCode: createExemptionCodeableConcepts(claimFormValues.exemption),
+    detail: medicationRequests.map((medicationRequest, index) => {
+      const lineItemId = getMedicationRequestLineItemId(medicationRequest)
+      const medicationDispensesForRequest = medicationDispenses.filter(
         medicationDispense => getMedicationDispenseLineItemId(medicationDispense) === lineItemId
       )
-      const productFormValuesForLineItem = claimFormValues.products.find(product => product.id === lineItemId)
+      const productFormValuesForRequest = claimFormValues.products.find(product => product.id === lineItemId)
       return createClaimItemDetail(
         index + 1,
-        lineItemId,
-        medicationRequestForLineItem,
-        medicationDispensesForLineItem,
-        productFormValuesForLineItem
+        medicationRequest,
+        medicationDispensesForRequest,
+        productFormValuesForRequest
       )
     })
   }
 }
 
+function createExemptionCodeableConcepts(exemption: ExemptionFormValues) {
+  const exemptionStatusCodeableConcept: fhir.CodeableConcept = {
+    coding: VALUE_SET_PRESCRIPTION_CHARGE_EXEMPTION.filter(coding => coding.code === exemption.code)
+  }
+
+  const evidenceSeenCodeableConcept = exemption.evidenceSeen
+    ? CODEABLE_CONCEPT_EXEMPTION_EVIDENCE_SEEN
+    : CODEABLE_CONCEPT_EXEMPTION_NO_EVIDENCE_SEEN
+
+  return [
+    exemptionStatusCodeableConcept,
+    evidenceSeenCodeableConcept
+  ]
+}
+
 function createClaimItemDetail(
   sequence: number,
-  lineItemId: string,
   medicationRequest: fhir.MedicationRequest,
   medicationDispenses: Array<fhir.MedicationDispense>,
   productFormValues: ProductFormValues
 ): fhir.ClaimItemDetail {
+  const lineItemId = getMedicationRequestLineItemId(medicationRequest)
```

Comment:
we make the same call in `createClaimItem` and we also stopped passing it down into detail, any reason why?

Reply:
We're already passing it in (in the MedicationRequest)